src/main/java/com/es/phoneshop/model/cart/DefaultCartService.java

```diff
@@ -0,0 +1,56 @@
+package com.es.phoneshop.model.cart;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.ProductDao;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.List;
+
+public class DefaultCartService implements CartService {
+    private static final String CART_SESSION_ATTRIBUTE = DefaultCartService.class.getName() + ".cart";
+    private ProductDao productDao;
+
+    private DefaultCartService() {
+        productDao = ArrayListProductDao.getInstance();
+    }
+
+    private static class SingleTonHelper {
+        private static final CartService INSTANCE = new DefaultCartService();
+    }
+
+    public static CartService getInstance() {
+        return DefaultCartService.SingleTonHelper.INSTANCE;
+    }
+
+    @Override
+    public synchronized Cart getCart(HttpServletRequest request) {
+        Cart cart = (Cart)request.getSession().getAttribute(CART_SESSION_ATTRIBUTE);
+        if (cart == null) {
+            request.getSession().setAttribute(CART_SESSION_ATTRIBUTE, cart = new Cart());
+        }
+        return cart;
+    }
+
+    @Override
+    public synchronized void add(Cart cart, Long productId, int quantity) throws OutOfStockException {
```

Comment:
Please refactor this method: 1) Create setter CartItem::setQuantity to use it for existing item to sum quantity instead of creating new CartItem object 2) Do not decrease stock of product, it's general approach in commerce applications, stock is decreased after user submits order